### PR TITLE
APPSEC-331: Update wc_sessionRequest domain regex matching

### PR DESCRIPTION
**Pull request by chriselbring-avalabs**
## Description

**Ticket: APPSEC-331**

Please provide:

- A summary of the changes
Anchors the core web regex and splits out localhost 
- Motivation and context for this change
Prevent non core domains from connecting in core context.
- Any dependencies introduced
No dependencies introduced
- (If breaking) migration steps and instructions

## Screenshots/Videos

Include relevant screenshots or screen recordings of iOS and Android.

## Testing

**Dev Testing (if applicable)**

- Provide steps to test the happy path of your feature
- Provide steps to test edge cases and error states
- Trigger a build on bitrise and reference it here
- Move the ticket into the "Testing" column on Jira

**QA Testing (if applicable)**

- Provide instructions for QA to test this feature thoroughly
- State expected behavior / acceptance criteria

## Checklist

Please check all that apply (if applicable)

- [ ] I have performed a self-review of my code
- [ ] I have verified the code works
- [ ] I have included screenshots / videos of android and ios
- [ ] I have added testing steps
- [ ] I have added/updated necessary unit tests
- [ ] I have updated the documentation

### CODE REVIEW

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/utils.ts [hunk 1]
```diff
@@ -74,9 +74,16 @@ export const isCoreDomain = (url: string): boolean => {
   const isCoreExt =
     CORE_EXT_HOSTNAMES.includes(hostname) && protocol === 'chrome-extension:'
 
+  const isHttps = protocol === 'https:'
+  const isLocalhost =
+    (hostname === 'localhost' || hostname === '127.0.0.1') &&
+    (protocol === 'http:' || protocol === 'https:')
```
Comment (atn4z7) [inline packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/utils.ts:80]: Minor cleanup: now that `isLocalhost` checks both http and https itself, the `'localhost'` and `'127.0.0.1'` entries up in `CORE_WEB_HOSTNAMES` are effectively dead. That array only gets checked inside the `isHttps` branch, and `isLocalhost` already covers the https case anyway, so they never do anything. Might be worth dropping them so the list is just the actual web hostnames. Not blocking.

**Review by atn4z7 — COMMENTED** [review]
Nice PR, change looks good to me.

**Review by atn4z7 — APPROVED** [review]

File: packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/utils.ts [hunk 2]
```diff
@@ -74,9 +74,16 @@ export const isCoreDomain = (url: string): boolean => {
...
+
   const isCoreWeb =
-    CORE_WEB_HOSTNAMES.includes(hostname) ||
-    CORE_WEB_URLS_REGEX.some(regex => new RegExp(regex).test(url))
+    isLocalhost ||
+    (isHttps &&
+      (CORE_WEB_HOSTNAMES.includes(hostname) ||
```
Comment (atn4z7) [inline packages/core-mobile/app/store/rpc/handlers/wc_sessionRequest/utils.ts:85]: Not from this PR, but while we're in here: `ava-labs.github.io` is matched as an exact hostname, so any repo or page published under the org's GitHub Pages gets treated as a Core domain, not just the playground. Since the ticket's about tightening over-trust, should we try to scope that down too? They playground url is https://ava-labs.github.io/extension-avalanche-playground/

**Review by onghwan — APPROVED** [review]
